### ISSUE TITLE
[BUG]: Duplicate questions being written to answers.json

### ISSUE THREAD

**Issue by sloganking** [feder-cr/Auto_Jobs_Applier_AIHawk#557]
### Describe the bug

Duplicates of the same question and answer pairs are being written to anwers.json

### Expected behavior

answers.json should store the same question only once.

### Actual behavior

answers.json stores a duplicate the same question every time the program encounters it

### Branch

main

### Python version

Python 3.12.0

### LLM Used

ChatGPT

### Model used

GPT-4o-mini

### Additional context

This problem occurs in multiple places. But an instanced of this problem occurring can be seen here

https://github.com/feder-cr/Auto_Jobs_Applier_AIHawk/blob/4f639023d3549894aa78a4566ae7a94fb4536a4f/src/aihawk_easy_applier.py#L714-L716

The

```python
if not is_cover_letter:
    self._save_questions_to_json({'type': question_type, 'question': question_text, 'answer': answer})
```

Does not check if an existing answer was found before saving the question and answer again. `existing_answer` is retrieved right before this if statement. So a fix to this could be 

```python
if not existing_answer and not is_cover_letter:
    self._save_questions_to_json({'type': question_type, 'question': question_text, 'answer': answer})
```

Or `self._save_questions_to_json` could check if it's parameter values are duplicate and it should not store them. But that would waste resources when we already know that an `existing_answer` was successfully retrieved and is not `None`.

**Comment by sloganking** [comment]
I am working on a PR for this.
